src/public/modules/forms/helpers/process-decrypted-content.ts

```diff
@@ -0,0 +1,95 @@
+import {
+  DecryptedContent,
+  FormField as VerifiedFormField,
+} from '@opengovsg/formsg-sdk/dist/types'
+import has from 'lodash/has'
+
+import {
+  CURRENT_VERIFIED_FIELDS,
+  VerifiedKeys,
+} from '../../../../shared/util/verified-content'
+import { BasicField, SPCPFieldTitle } from '../../../../types'
+
+/**
+ * Returns a response matching the given type containing the given value.
+ * @param type the field type to match
+ * @param value the value to insert into the response to be returned
+ * @returns the desired response object if type is valid. Else returns null.
+ */
+const getResponseFromVerifiedField = (
+  type: VerifiedKeys,
+  value: string,
+): VerifiedFormField | null => {
+  switch (type) {
+    case VerifiedKeys.SpUinFin:
+      return {
+        question: SPCPFieldTitle.SpNric,
+        fieldType: BasicField.Nric,
+        answer: value,
+        // Just a unique identifier for CSV header uniqueness
+        _id: SPCPFieldTitle.SpNric,
+      }
+
+    case VerifiedKeys.CpUen:
+      return {
+        question: SPCPFieldTitle.CpUen,
+        fieldType: BasicField.ShortText,
+        answer: value,
+        _id: SPCPFieldTitle.CpUen,
+      }
+
+    case VerifiedKeys.CpUid:
+      return {
+        question: SPCPFieldTitle.CpUid,
+        fieldType: BasicField.Nric,
+        answer: value,
+        _id: SPCPFieldTitle.CpUid,
+      }
+    default:
+      return null
+  }
+}
+
+const isVerifiedFormField = (
+  possibleFormField: unknown,
+): possibleFormField is VerifiedFormField => {
+  return !!possibleFormField && !!(possibleFormField as VerifiedFormField)._id
```

Comment:
may want to use lodash's `get` just in case we are unable to destructure, instead of casting.

```ts
return !!possibleFormField && !!get(possibleFormField, _id)
```

can we also do stronger checks for what exactly constitutes a verified form field?

I see the typing in the sdk is

```ts

    _id: string;
    question: string;
    fieldType: FieldType;
    isHeader?: boolean;
    signature?: string;
} & ({
    answer: string;
    answerArray?: never;
} | {
    answer?: never;
    answerArray: string[] | string[][];
});
```

can i also query on why we need this typeguard? if we are using SDK types, should we not be assured that if the decrypted response is not null, then it will be a "VerifiedFormField"?

Reply:
we need the typeguard because we are trying to transform the type from `VerifiedField | null -> VerifiedField`. Oddly enough, ts can't infer the type if i do `.filter(field => !!field)` and complains that `Signature '(field: FormField | null): boolean' must be a type predicate`

wrt the checks, you're right in that if the response is non-null then it'll be a `VerifiedFormField`; the reason why i did it as such as cos i set the initial type to unknown but i think it's a premature optimization and i'll just type it according to the sdk HAHAH